lib/chore/queues/sqs/consumer.rb

```diff
@@ -88,10 +88,11 @@ def delay(item, backoff_calc)
         def handle_messages(&block)
           msg = queue.receive_messages(:max_number_of_messages => sqs_polling_amount, :attribute_names => ['ApproximateReceiveCount'])
           messages = *msg
+          received_timestamp = Time.now
 
           messages.each do |message|
-            unless duplicate_message?(message.message_id, message.queue_url, queue_timeout)
-              block.call(message.message_id, message.receipt_handle, queue_name, queue_timeout, message.body, message.attributes['ApproximateReceiveCount'].to_i - 1)
+            unless duplicate_message?(message.message_id, message.queue_url, queue_timeout, received_timestamp)
+              block.call(message.message_id, message.receipt_handle, queue_name, queue_timeout, message.body, message.attributes['ApproximateReceiveCount'].to_i - 1, received_timestamp)
```

Comment:
New Implementation 5/20: Capture `received_timestamp` and pass it into both the `consumer` and `workers`'s duplicate handling logic. Keeps `received_timestamp` unified/the same value 

(1/5)